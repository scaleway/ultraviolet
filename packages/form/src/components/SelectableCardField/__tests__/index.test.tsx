import { act, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm } from '@utils/test'
import { describe, expect, test, vi } from 'vitest'
import { SelectableCardField } from '../..'

describe('SelectableCardField', () => {
  test('should render correctly', () => {
    const { asFragment } = renderWithForm(
      <SelectableCardField label="test" name="test" value="test">
        Radio field
      </SelectableCardField>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly disabled', () => {
    const { asFragment } = renderWithForm(
      <SelectableCardField disabled label="test" name="test" value="disabled">
        Radio field disabled
      </SelectableCardField>,
    )
    const input = screen.getByRole('radio', { hidden: true })
    expect(input).toBeDisabled()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly checked', () => {
    const { asFragment } = renderWithForm(
      <SelectableCardField label="test" name="test" value="checked">
        Radio field checked
      </SelectableCardField>,
      { defaultValues: { test: 'checked' } },
    )
    const input = screen.getByRole('radio', { hidden: true })
    expect(input).toBeChecked()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should trigger events correctly', async () => {
    const onFocus = vi.fn(() => {})
    const onChange = vi.fn(() => {})
    const onBlur = vi.fn(() => {})

    const { asFragment } = renderWithForm(
      <SelectableCardField
        label="test"
        name="test"
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        value="events"
      >
        Radio field events
      </SelectableCardField>,
    )
    const input = screen.getByRole('radio', { hidden: true })
    act(() => input.focus())
    expect(onFocus).toBeCalledTimes(1)
    await userEvent.click(input)
    expect(onChange).toBeCalledTimes(1)
    act(() => input.blur())
    expect(onBlur).toBeCalledTimes(1)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly with errors', async () => {
    const { asFragment } = renderWithForm(
      <>
        <SelectableCardField label="test" name="test" required value="checked">
          Radio field error
        </SelectableCardField>
        <button type="submit">Submit</button>
      </>,
    )
    await userEvent.click(screen.getAllByRole('button')[1])
    const input = screen.getByRole('radio', { hidden: true })
    expect(input).toHaveAttribute('aria-invalid', 'true')
    expect(asFragment()).toMatchSnapshot()
  })
})
