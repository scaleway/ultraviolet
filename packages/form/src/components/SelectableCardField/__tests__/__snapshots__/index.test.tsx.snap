// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`selectableCardField > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
}

.emotion-0[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-0[data-has-default-cursor='true'] {
  cursor: default;
}

.emotion-0[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-0[data-image="illustration"] {
  padding: 0rem;
}

.emotion-0[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-0:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-0:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-0[data-has-label='true'] uv_17da2jl1,
.emotion-0[data-has-label='true'] .eqr7bqq0 {
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-2[data-checked='true'] {
  color: #641cb3;
}

.emotion-2[data-error='true'] {
  color: #b3144d;
}

.emotion-2[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-2 input+svg {
  display: none;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 {
  fill: #d9dadd;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 {
  fill: #8c40ef;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 {
  background: none;
  fill: #8c40ef;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-4[data-has-label='true'] {
  padding-left: 2rem;
}

.emotion-4[data-has-label='false'] {
  display: contents;
}

.emotion-4[data-has-default-cursor='true'] {
  cursor: default;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1 uv_toi52u0 uv_toi52u3j uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        data-checked="false"
        data-disabled="false"
        data-error="false"
        data-has-default-cursor="false"
        data-has-label="true"
        data-image="none"
        data-type="radio"
        role="button"
        style="--uv_4k0ekn3: 1;"
        tabindex="0"
      >
        <div
          class="uv_17da2jl1 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        >
          <div
            aria-disabled="false"
            class="emotion-2 emotion-3 uv_17da2jl3"
            data-checked="false"
            data-error="false"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              class="uv_17da2jl6"
              id="«r1»"
              name="test"
              tabindex="-1"
              type="radio"
              value="test"
            />
            <svg
              class="uv_17da2jl7"
              viewBox="0 0 24 24"
            >
              <g>
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                  stroke-width="2"
                />
                <circle
                  class="uv_17da2jl0"
                  cx="12"
                  cy="12"
                  r="8"
                />
                <circle
                  class="uv_17da2jl8"
                  cx="12"
                  cy="12"
                  r="5"
                />
              </g>
            </svg>
            <label
              class="uv_17da2jl4 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
              for="«r1»"
            >
              test
            </label>
          </div>
        </div>
        <div
          class="emotion-4 emotion-5 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
          data-has-default-cursor="false"
          data-has-label="false"
          style="--uv_4k0ekn0: 100%;"
        >
          Radio field
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectableCardField > should render correctly checked 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
}

.emotion-0[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-0[data-has-default-cursor='true'] {
  cursor: default;
}

.emotion-0[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-0[data-image="illustration"] {
  padding: 0rem;
}

.emotion-0[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-0:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-0:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-0[data-has-label='true'] uv_17da2jl1,
.emotion-0[data-has-label='true'] .eqr7bqq0 {
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-2[data-checked='true'] {
  color: #641cb3;
}

.emotion-2[data-error='true'] {
  color: #b3144d;
}

.emotion-2[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-2 input+svg {
  display: none;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 {
  fill: #d9dadd;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 {
  fill: #8c40ef;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 {
  background: none;
  fill: #8c40ef;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-4[data-has-label='true'] {
  padding-left: 2rem;
}

.emotion-4[data-has-label='false'] {
  display: contents;
}

.emotion-4[data-has-default-cursor='true'] {
  cursor: default;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1 uv_toi52u0 uv_toi52u3j uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        data-checked="true"
        data-disabled="false"
        data-error="false"
        data-has-default-cursor="false"
        data-has-label="true"
        data-image="none"
        data-type="radio"
        role="button"
        style="--uv_4k0ekn3: 1;"
        tabindex="0"
      >
        <div
          class="uv_17da2jl1 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        >
          <div
            aria-disabled="false"
            class="emotion-2 emotion-3 uv_17da2jl3"
            data-checked="true"
            data-error="false"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              checked=""
              class="uv_17da2jl6"
              id="«r9»"
              name="test"
              tabindex="-1"
              type="radio"
              value="checked"
            />
            <svg
              class="uv_17da2jl7"
              viewBox="0 0 24 24"
            >
              <g>
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                  stroke-width="2"
                />
                <circle
                  class="uv_17da2jl0"
                  cx="12"
                  cy="12"
                  r="8"
                />
                <circle
                  class="uv_17da2jl8"
                  cx="12"
                  cy="12"
                  r="5"
                />
              </g>
            </svg>
            <label
              class="uv_17da2jl4 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
              for="«r9»"
            >
              test
            </label>
          </div>
        </div>
        <div
          class="emotion-4 emotion-5 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
          data-has-default-cursor="false"
          data-has-label="false"
          style="--uv_4k0ekn0: 100%;"
        >
          Radio field checked
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectableCardField > should render correctly disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
}

.emotion-0[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-0[data-has-default-cursor='true'] {
  cursor: default;
}

.emotion-0[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-0[data-image="illustration"] {
  padding: 0rem;
}

.emotion-0[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-0:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-0:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-0[data-has-label='true'] uv_17da2jl1,
.emotion-0[data-has-label='true'] .eqr7bqq0 {
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-2[data-checked='true'] {
  color: #641cb3;
}

.emotion-2[data-error='true'] {
  color: #b3144d;
}

.emotion-2[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-2 input+svg {
  display: none;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 {
  fill: #d9dadd;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 {
  fill: #8c40ef;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 {
  background: none;
  fill: #8c40ef;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-4[data-has-label='true'] {
  padding-left: 2rem;
}

.emotion-4[data-has-label='false'] {
  display: contents;
}

.emotion-4[data-has-default-cursor='true'] {
  cursor: default;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1 uv_toi52u0 uv_toi52u3j uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        data-checked="false"
        data-disabled="true"
        data-error="false"
        data-has-default-cursor="false"
        data-has-label="true"
        data-image="none"
        data-type="radio"
        role="button"
        style="--uv_4k0ekn3: 1;"
      >
        <div
          class="uv_17da2jl1 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        >
          <div
            aria-disabled="true"
            class="emotion-2 emotion-3 uv_17da2jl3"
            data-checked="false"
            data-error="false"
          >
            <input
              aria-disabled="true"
              aria-invalid="false"
              class="uv_17da2jl6"
              disabled=""
              id="«r5»"
              name="test"
              tabindex="-1"
              type="radio"
              value="disabled"
            />
            <svg
              class="uv_17da2jl7"
              viewBox="0 0 24 24"
            >
              <g>
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                  stroke-width="2"
                />
                <circle
                  class="uv_17da2jl0"
                  cx="12"
                  cy="12"
                  r="8"
                />
                <circle
                  class="uv_17da2jl8"
                  cx="12"
                  cy="12"
                  r="5"
                />
              </g>
            </svg>
            <label
              class="uv_17da2jl4 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
              for="«r5»"
            >
              test
            </label>
          </div>
        </div>
        <div
          class="emotion-4 emotion-5 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
          data-has-default-cursor="false"
          data-has-label="false"
          style="--uv_4k0ekn0: 100%;"
        >
          Radio field disabled
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectableCardField > should render correctly with errors 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
}

.emotion-0[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-0[data-has-default-cursor='true'] {
  cursor: default;
}

.emotion-0[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-0[data-image="illustration"] {
  padding: 0rem;
}

.emotion-0[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-0:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-0:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-0[data-has-label='true'] uv_17da2jl1,
.emotion-0[data-has-label='true'] .eqr7bqq0 {
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-2[data-checked='true'] {
  color: #641cb3;
}

.emotion-2[data-error='true'] {
  color: #b3144d;
}

.emotion-2[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-2 input+svg {
  display: none;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 {
  fill: #d9dadd;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 {
  fill: #8c40ef;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 {
  background: none;
  fill: #8c40ef;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-4[data-has-label='true'] {
  padding-left: 2rem;
}

.emotion-4[data-has-label='false'] {
  display: contents;
}

.emotion-4[data-has-default-cursor='true'] {
  cursor: default;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1 uv_toi52u0 uv_toi52u3j uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        data-checked="false"
        data-disabled="false"
        data-error="true"
        data-has-default-cursor="false"
        data-has-label="true"
        data-image="none"
        data-type="radio"
        role="button"
        style="--uv_4k0ekn3: 1;"
        tabindex="0"
      >
        <div
          class="uv_17da2jl1 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        >
          <div
            aria-disabled="false"
            class="emotion-2 emotion-3 uv_17da2jl3"
            data-checked="false"
            data-error="true"
          >
            <input
              aria-disabled="false"
              aria-invalid="true"
              class="uv_17da2jl6"
              id="«rh»"
              name="test"
              tabindex="-1"
              type="radio"
              value="checked"
            />
            <svg
              class="uv_17da2jl7"
              viewBox="0 0 24 24"
            >
              <g>
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                  stroke-width="2"
                />
                <circle
                  class="uv_17da2jl0"
                  cx="12"
                  cy="12"
                  r="8"
                />
                <circle
                  class="uv_17da2jl8"
                  cx="12"
                  cy="12"
                  r="5"
                />
              </g>
            </svg>
            <label
              class="uv_17da2jl4 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
              for="«rh»"
            >
              test
            </label>
          </div>
        </div>
        <div
          class="emotion-4 emotion-5 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
          data-has-default-cursor="false"
          data-has-label="false"
          style="--uv_4k0ekn0: 100%;"
        >
          Radio field error
        </div>
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectableCardField > should trigger events correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  padding: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: border-color 200ms ease,box-shadow 200ms ease;
  transition: border-color 200ms ease,box-shadow 200ms ease;
  cursor: pointer;
  background: #ffffff;
  border: 1px solid #d9dadd;
  color: #3f4250;
}

.emotion-0[data-has-label='false']>:first-child {
  margin-bottom: -0.25rem;
}

.emotion-0[data-has-default-cursor='true'] {
  cursor: default;
}

.emotion-0[data-checked='true'] {
  border: 1px solid #8c40ef;
}

.emotion-0[data-error='true'] {
  border: 1px solid #b3144d;
}

.emotion-0[data-disabled='true'] {
  border: 1px solid #e9eaeb;
  color: #b5b7bd;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-0[data-image="illustration"] {
  padding: 0rem;
}

.emotion-0[data-image="icon"] {
  padding: 0rem;
  padding-right: 1rem;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true']),
.emotion-0:active:not([data-error='true']):not([data-disabled='true']) {
  border: 1px solid #8c40ef;
}

.emotion-0:hover:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'],
.emotion-0:active:not([data-error='true']):not([data-disabled='true'])[data-cheked='false'] {
  box-shadow: 0px 4px 16px 4px #f6f3ffcc;
}

.emotion-0[data-has-label='true'] uv_17da2jl1,
.emotion-0[data-has-label='true'] .eqr7bqq0 {
  width: 100%;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-2[data-checked='true'] {
  color: #641cb3;
}

.emotion-2[data-error='true'] {
  color: #b3144d;
}

.emotion-2[aria-disabled='true'] {
  color: #b5b7bd;
}

.emotion-2 input+svg {
  display: none;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 {
  fill: #d9dadd;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false']:not([data-checked='true']) uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 {
  fill: #8c40ef;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 {
  fill: #b3144d;
}

.emotion-2:hover[aria-disabled='false'] uv_17da2jl6[aria-invalid='true']+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 {
  background: none;
  fill: #8c40ef;
}

.emotion-2 uv_17da2jl6[aria-disabled='false']:active+uv_17da2jl7 uv_17da2jl0 {
  fill: #ffffff;
}

.emotion-4[data-has-label='true'] {
  padding-left: 2rem;
}

.emotion-4[data-has-label='false'] {
  display: contents;
}

.emotion-4[data-has-default-cursor='true'] {
  cursor: default;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        class="emotion-0 emotion-1 uv_toi52u0 uv_toi52u3j uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        data-checked="true"
        data-disabled="false"
        data-error="false"
        data-has-default-cursor="false"
        data-has-label="true"
        data-image="none"
        data-type="radio"
        role="button"
        style="--uv_4k0ekn3: 1;"
        tabindex="0"
      >
        <div
          class="uv_17da2jl1 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
        >
          <div
            aria-disabled="false"
            class="emotion-2 emotion-3 uv_17da2jl3"
            data-checked="true"
            data-error="false"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              class="uv_17da2jl6"
              id="«rd»"
              name="test"
              tabindex="-1"
              type="radio"
              value="events"
            />
            <svg
              class="uv_17da2jl7"
              viewBox="0 0 24 24"
            >
              <g>
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                  stroke-width="2"
                />
                <circle
                  class="uv_17da2jl0"
                  cx="12"
                  cy="12"
                  r="8"
                />
                <circle
                  class="uv_17da2jl8"
                  cx="12"
                  cy="12"
                  r="5"
                />
              </g>
            </svg>
            <label
              class="uv_17da2jl4 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
              for="«rd»"
            >
              test
            </label>
          </div>
        </div>
        <div
          class="emotion-4 emotion-5 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
          data-has-default-cursor="false"
          data-has-label="false"
          style="--uv_4k0ekn0: 100%;"
        >
          Radio field events
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
